packages/labs/ssr/src/lib/lit-element-renderer.ts

```diff
@@ -29,6 +33,31 @@ export class LitElementRenderer extends ElementRenderer {
   constructor(tagName: string) {
     super(tagName);
     this.element = new (customElements.get(this.tagName)!)() as LitElement;
+
+    // Reflect internals AOM attributes back to the DOM prior to hydration to
+    // ensure search bots can accurately parse element semantics prior to
+    // hydration. This is called whenever an instance of ElementInternals is
+    // created on an element to wire up the getters/setters for the ARIAMixin
+    // properties.
+    const internals = (
+      this.element as object as {__internals: ElementInternals}
+    ).__internals;
+    if (internals) {
+      for (const [key, value] of Object.entries(internals)) {
```

Comment:
I would iterate on the enum. The internal object could conceivably have an accessor which wouldn't be returned by Object.entries().

Reply:
Done